### PR TITLE
I98 gitlab install

### PR DESCRIPTION
**Pull request by kayiwa**
PR installs local ldap enabled gitlab instance for private repos. 
Related to https://github.com/pulibrary/ops-catchall/issues/84

### CODE REVIEW

File: inventory/all_projects/pulgit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[pulgit_production]
```
Comment (acozine) [inline inventory/all_projects/pulgit:1]: Can we call the VMs and groups `pulgitlab` or something? I feel like `git` is too generic, I'm not sure if it's a platform like github/gitlab, or some kind of git practice thing, or what.

File: group_vars/gitlab/production.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+gitlab_pu_ldap_bind_dn: "{{ vault_gitlab_pu_ldap_bind_dn }}"
+gitlab_pu_ldap_password: "{{ vault_gitlab_ldap_password }}"
+gitlab_loadbalancer_domain_name: "pulgit.lib.princeton.edu"
```
Comment (acozine) [inline group_vars/gitlab/production.yml:4]: I would call this `gitlab.lib.princeton.edu`, I think - no reason to repeat the library part.

File: inventory/all_projects/pulgit [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+git.lib.princeton.edu
+[pulgit_staging]
+git-staging.lib.princeton.edu
```
Comment (acozine) [inline inventory/all_projects/pulgit:4]: What is the purpose of the staging system? Is it just to test updates to the GitLab software?

File: playbooks/git.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+---
+# by default this playbook runs in the staging environment
+# to run in production, pass '-e runtime_env=production'
+
+- name: build the git vcs
```
Comment (acozine) [inline playbooks/git.yml:5]:
```suggestion
- name: build our in-house GitLab instances
```

File: playbooks/git.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  hosts: pulgit_{{ runtime_env | default('staging') }}
```
Comment (acozine) [inline playbooks/git.yml:6]: If we change the group names, we would need to change them here too.

File: roles/nginxplus/files/conf/http/dev/pulgit_staging.conf [hunk 6]
```diff
@@ -0,0 +1,42 @@
+# Ansible managed
+proxy_cache_path /var/cache/nginx/pulgit-staging/ keys_zone=pulgitstagingcache:10m;
+
+upstream pulgit-staging {
```
Comment (acozine) [inline roles/nginxplus/files/conf/http/dev/pulgit_staging.conf:4]: Name change would need to happen throughout this file too.

**Review by acozine — COMMENTED** [review]
This is a lot of work and a great step forward.

I'd like to see consistency in naming - right now some elements are named `git_x` and others are `pulgit_x` and others are `gitlab_x`. Why not make it easy on ourselves and call everything (inventory, VM names, DNS, role, etc., etc.) `gitlab`? It's clear and consistent and describes precisely what the system does.

File: inventory/all_projects/pulgit (hunk 3 above)
Comment (kayiwa) [reply inventory/all_projects/pulgit:4]: I have done the migration from svn -> git before. I will want more than one person to verify this process. Testing this on staging will make me feel better.

File: playbooks/git.yml (hunk 4 above)
Comment (acozine) [reply playbooks/git.yml:5]: I would still like to see this name changed. When I see `build the git vcs` I don't have an immediate understanding of what is happening.

File: roles/example/molecule/default/converge.yml [hunk 7]
```diff
@@ -5,11 +5,11 @@
     - running_on_server: false
   become: true
   pre_tasks:
-    - name: update cache
-      apt:
+    - name: Update cache
+      ansible.builtin.apt:
```
Comment (acozine) [inline roles/example/molecule/default/converge.yml:9]: Nice!

File: playbooks/git.yml (hunk 4 above)
Comment (acozine) [reply playbooks/git.yml:5]: I would also like to see the name of the playbook file changed from `git.yml` to `GitLab.yml`.

File: inventory/all_projects/gitlab [hunk 8]
```diff
@@ -0,0 +1,4 @@
+[gitlab_production]
+git.lib.princeton.edu
```
Comment (acozine) [inline inventory/all_projects/gitlab:2]: I would love to see the VM names changed also - gitlab.lib.princeon.edu and gitlab-staging.lib.princeton.edu.

File: roles/nginxplus/files/conf/http/dev/gitlab_staging.conf [hunk 9]
```diff
@@ -0,0 +1,42 @@
+# Ansible managed
+proxy_cache_path /var/cache/nginx/gitlab-staging/ keys_zone=gitlabstagingcache:10m;
+
+upstream gitlab-staging {
+    zone gitlab-staging 64k;
+    server git-staging.lib.princeton.edu:443 resolve;
```
Comment (acozine) [inline roles/nginxplus/files/conf/http/dev/gitlab_staging.conf:6]: VM name would also need to change here.

File: roles/nginxplus/files/conf/http/dev/gitlab_staging.conf [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    sticky learn
+          create=$upstream_cookie_gitlabstagingcookie
+          lookup=$cookie_gitlabstagingcookie
+          zone=gitlabstagingclient_sessions:1m;
+}
+
+server {
+    listen 80;
+    server_name gitlab-staging.lib.princeton.edu;
```
Comment (acozine) [inline roles/nginxplus/files/conf/http/dev/gitlab_staging.conf:15]: and here

File: roles/nginxplus/files/conf/http/dev/gitlab_staging.conf [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+    location / {
+        return 301 https://$server_name$request_uri;
+    }
+}
+
+server {
+    listen 443 ssl;
+    http2 on;
+    server_name gitlab-staging.lib.princeton.edu;
```
Comment (acozine) [inline roles/nginxplus/files/conf/http/dev/gitlab_staging.conf:25]: and here

File: roles/nginxplus/files/conf/http/gitlab_prod.conf [hunk 12]
```diff
@@ -0,0 +1,42 @@
+# Ansible managed
+proxy_cache_path /data/nginx/gitlab/NGINX_cache/ keys_zone=gitlabcache:10m;
+
+upstream gitlab {
+    zone gitlab 64k;
+    server git.lib.princeton.edu:443 resolve;
```
Comment (acozine) [inline roles/nginxplus/files/conf/http/gitlab_prod.conf:6]: prod server name would need to change here

File: roles/nginxplus/files/conf/http/gitlab_prod.conf [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+    sticky learn
+          create=$upstream_cookie_gitlabcookie
+          lookup=$cookie_gitlabcookie
+          zone=gitlabclient_sessions:1m;
+}
+
+server {
+    listen 80;
+    server_name gitlab.lib.princeton.edu;
```
Comment (acozine) [inline roles/nginxplus/files/conf/http/gitlab_prod.conf:15]: and here

File: roles/nginxplus/files/conf/http/gitlab_prod.conf [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+    location / {
+        return 301 https://$server_name$request_uri;
+    }
+}
+
+server {
+    listen 443 ssl;
+    http2 on;
+    server_name gitlab.lib.princeton.edu;
```
Comment (acozine) [inline roles/nginxplus/files/conf/http/gitlab_prod.conf:25]: and here

**Review by acozine — COMMENTED** [review]
Ugh, I missed a few naming things on my first review.

Basically, it would be great to have everything use the `gitlab` name, instead of `git` - because the role doesn't build `git`, it builds a GitLab instance.

File: inventory/all_projects/gitlab (hunk 8 above)
Comment (kayiwa) [reply inventory/all_projects/gitlab:2]: does gitlab-stagingvm and gitlab-prodvm work

Comment (acozine) [inline inventory/all_projects/gitlab:2]:
```suggestion
gitlab-prod-vm.lib.princeton.edu
```

File: inventory/all_projects/gitlab [hunk 15]
```diff
@@ -0,0 +1,4 @@
...
+[gitlab_staging]
+git-staging.lib.princeton.edu
```
Comment (acozine) [inline inventory/all_projects/gitlab:4]:
```suggestion
gitlab-staging-vm.lib.princeton.edu
```

File: roles/nginxplus/files/conf/http/dev/gitlab_staging.conf (hunk 9 above)
Comment (acozine) [inline roles/nginxplus/files/conf/http/dev/gitlab_staging.conf:6]:
```suggestion
    server gitlab-staging-vm.lib.princeton.edu:443 resolve;
```

File: roles/nginxplus/files/conf/http/gitlab_prod.conf (hunk 12 above)
Comment (acozine) [inline roles/nginxplus/files/conf/http/gitlab_prod.conf:6]:
```suggestion
    server gitlab-prod-vm.lib.princeton.edu:443 resolve;
```

**Review by acozine — COMMENTED** [review]
After discussion, adding these as suggestions.

File: inventory/all_projects/gitlab (hunk 8 above)
Comment (acozine) [reply inventory/all_projects/gitlab:2]: Yep, see suggestions below.

**Pull request merged**